### PR TITLE
Remove auth registry credentials

### PR DESCRIPTION
**Pull request by jesusbv**

### CODE REVIEW

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [inline lib/cloudregister/registerutils.py:603]: this `try/except` block of updating the `/etc/containers/config.json` file can has its own method so it can be reused in `set_registry_auth_token()` as well

I plan to do that once the registry PRs are merged

Comment (jesusbv) [reply lib/cloudregister/registerutils.py:602]: Created the method in #159, no reason to wait

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:634]: We should check if our content is the only content in the file and if it is simply `unlink` the file.

File: tests/test_registerutils.py
Comment (rjschwei) [inline tests/test_registerutils.py:3607]: s/toke/token/

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:636]: If our entry was the only entry config_json contains no useful information and we'll write `{'auths':{}}` which is not meaningful.

File: lib/cloudregister/registerutils.py
Comment (schaefi) [inline lib/cloudregister/registerutils.py:546]: This is not catching opening errors, e.g permission denied. Intentional ?

File: lib/cloudregister/registerutils.py
Comment (schaefi) [inline lib/cloudregister/registerutils.py:562]: We need our own exec_subprocess to move a file ? How about ```shutil.move```. It's also unclear to me why the selinux fattr needs to be reset to the default. It actually feels wrong to do that if there is a selinux context set on the file. If this is wanted the comment should rather state the reason for the use of the mv options instead of saying in text what the next line does :)

File: lib/cloudregister/registerutils.py
Comment (schaefi) [inline lib/cloudregister/registerutils.py:585]: I would use ```set_new=True``` for better readability

File: lib/cloudregister/registerutils.py
Comment (schaefi) [reply lib/cloudregister/registerutils.py:634]: I would also delete the file instead of changing the content. The other logic for creating the credentials is also based on the presence of the file and iirc the logic above could leave a valid file with empty auths which confuses the creation of a new credentials file

**Review by schaefi — COMMENTED** [review]
Thanks for your work :+1: I think the logic in remove_auth_entry needs to be refactored. Other than that only nitpick on error handling and readability :)

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:562]: > We need our own exec_subprocess to move a file ? How about shutil.move

the suggested method to use was `os.system` and call a cmd, happy to use `shutil.move`, 
if we are calling a cmd, yes  it is my opinion to use the same method for that `exec_subprocess` in this case

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:546]: Yes, it will get the [code](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L543) from #159 once merged into this branch

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:636]: good catch, was treating `config_json` as if `auths` had content

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: if the file has other content, I would not remove the file

The file gets removed when
- empty content -> `{}`
- not relevant content -> `{"auths": {}`
- only our entry -> `{"auths": {our_fqdn: our_auth_token}`

otherwise, files does not get removed

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:555]: master branch merged and conflicts fixed, tests updated

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:540]: The function is called `get_registry_credentials()` in this case we cannot return the credentials, so I would consider that a failure, but we return `None` maybe indicating no failure? The intend here is not clear and we need to sort out what it should be.

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [reply lib/cloudregister/registerutils.py:571]: `os.system` was an example, anyway I agree we already have a subprocess wrapper we should use it.

With respect to the SELinux context, I am not sure how that works when we create the file or if a user created the file. We get here when we cannot read the existing config file, for whatever reason and since we create a backup maybe we do not care for the SELinux context. That said, this was part of the previous PR and is an artifact in this diff because this branch has not yet been rebased. An artifact of chaining branches and then not waiting until the start of the chain is merged and rebasing before sending a new PR. Which makes review more difficult.

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:675]: s/worht/worth/

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:653]: s/The content of the registry auth is our credentials only./Registry authentication config only contains managed content/

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:571]: I applied the changes from master on this branch yesterday

> Which makes review more difficult

Yes, sorry about that, it's a side effect of the time constraint we have for this feature
Hopefully, it is not that much more difficult

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [inline lib/cloudregister/registerutils.py:668]: I think in [this case,](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/162/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R696) checking the content on a dict like `{"auths": {"xxxxx?": "auth_token_we_can_match"}}`

I think for this case, it would be better to check the parts we know of the FQDN, will modify it for that

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:540]: Yes, the returning of `None` is intentional to differentiate the case of the non existing file with the case of existing file and failing and failing to open it or accessing the file and being empty, for example

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: Done

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: Done

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: Moved the check outside the method, I think that makes the intent clear

File: lib/cloudregister/registerutils.py
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:567]: > anyway I agree we already have a subprocess wrapper we should use it.

@rjschwei I understand you prefer `exec_subprocess` than using `os.rename` for this case, correct ?
or maybe not prefer as a strong word but maybe more "let's make fewer changes"

Comment (schaefi) [reply lib/cloudregister/registerutils.py:567]: I'm fine either way. If you still need the SELinux context related move, let's stick to how it is right now

**Review by schaefi — APPROVED** [review]
From my perspective this looks good

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:547]: Something is not right here, either the merge didn't work or the diff is off, what is shown as the diff that we try to read the config file and if we cannot then we log a message that we could not preserve the file. Given such inconsistencies it is basically impossible to review the PR.

Comment (jesusbv) [reply lib/cloudregister/registerutils.py:547]: I see
```python
    except OSError as error:
        message = (
            'Unable to open %s: %s, preserving file as %s.bak'
            % (REGISTRY_CREDENTIALS_PATH, error, REGISTRY_CREDENTIALS_PATH)
        )
```
which is part of the original intent
open the credentials file to update the credentials, 
opening failed, preserve the file <-- that is the part of that exception
create a new net credentials file with the info
  
 I can squash the commits into one, see if that help ?

Comment (rjschwei) [reply lib/cloudregister/registerutils.py:547]: But the preserving of the file is not there, line 547 in the diff has a `)` from the log message, then line 548 has `if set_new`. There is no attempt to create `.bak`

Comment (jesusbv) [reply lib/cloudregister/registerutils.py:547]: yes, there is ? if there was an [error](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/162/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R560), we do the preserve-file op

Comment (rjschwei) [reply lib/cloudregister/registerutils.py:547]: Ok there has to be a way to handle this with reasonable code locality.

Comment (jesusbv) [reply lib/cloudregister/registerutils.py:547]: we do the attempt to move the file when there are two exceptions, so we do it once instead of twice and log the message accordingly, we have used that style in our code base before

File: lib/cloudregister/registerutils.py
Comment (rjschwei) [inline lib/cloudregister/registerutils.py:564]: Since `failed` has 3 states, which does not make sense for a variable named `failed` we should rename it to `status`.

**Review by rjschwei — CHANGES_REQUESTED** [review]

**Review by rjschwei — APPROVED** [review]
Please squash before merging.

**Pull request merged**
